### PR TITLE
Make --namespace persisting option 4423

### PR DESCRIPTION
**Pull request by kirushik**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
@digitaltom will review your pull request :dancers: check https://trello.com/c/vEScqycd/4423-l3-suseconnect-doesnt-set-namespace-parameter
  @digitaltom: Please review this pull request using our guidelines:
  * test for acceptance criteria / functionality
  * check if the new code is covered with tests
  * check for unintended consequences
  * encourage usage of the boyscout rule
  * make sure the code is architected in the best way
  * check that no unnecessary technical debt got introduced
  * make sure that no unnecessary FIXMEs or TODOs got added

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/24419/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 448ccccca8aa1279a39495bc53c41776f7fdd480 into 1d32a31ced1a47f21a768e222ed24f1386c9b2b4 )

File: features/namespace.feature [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    When there's a file "/etc/SUSEConnect" with a line "namespace: foobar"
+    When I call SUSEConnect with '--regcode VALID --write-config true' arguments
+    Then a file named "/etc/SUSEConnect" should exist
+    And the file "/etc/SUSEConnect" should contain "namespace: foobar"
+
+  Scenario: It overrides and save namespace from CLI arguments
+    When there's a file "/etc/SUSEConnect" with a line "namespace: foobar"
+    When I call SUSEConnect with '--regcode VALID --namespace barbaz' arguments
+    Then a file named "/etc/SUSEConnect" should exist
+    And the file "/etc/SUSEConnect" should contain "namespace: barbaz"
```
Comment (thutterer) [inline features/namespace.feature]: No newline at end of file

File: features/namespace.feature [hunk 2]
```diff
@@ -0,0 +1,19 @@
+@slow_process
+Feature: `namespace` CLI argument and option persistence
+
+  Scenario: It stores namespace from CLI arguments argument in the config file
```
Comment (thutterer) [inline features/namespace.feature]: "arguments argument" ... one word too many?

File: features/namespace.feature [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    Then a file named "/etc/SUSEConnect" should exist
+    And the file "/etc/SUSEConnect" should contain "namespace: foobar"
+
+  Scenario: It keeps namespace from a configfile when no parameter specified
+    When there's a file "/etc/SUSEConnect" with a line "namespace: foobar"
+    When I call SUSEConnect with '--regcode VALID --write-config true' arguments
+    Then a file named "/etc/SUSEConnect" should exist
+    And the file "/etc/SUSEConnect" should contain "namespace: foobar"
+
+  Scenario: It overrides and save namespace from CLI arguments
```
Comment (thutterer) [inline features/namespace.feature]: "saves" :man_teacher:

File: features/namespace.feature [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    When there's a file "/etc/SUSEConnect" with a line "namespace: foobar"
+    When I call SUSEConnect with '--regcode VALID --namespace barbaz' arguments
```
Comment (thutterer) [inline features/namespace.feature]: Seems you could not decide if you like single- or double quotes better :wink:

**Comment by kirushik** [comment]
Please merge only after #298 is merged and verified to pass the Maintenance

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/24429/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged aaec0169fb97e814f315ddb934e3895a37d4f8c6 into 1d32a31ced1a47f21a768e222ed24f1386c9b2b4 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/24654/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged aaec0169fb97e814f315ddb934e3895a37d4f8c6 into dee6a7d6b9465dbf0d965804a55bb61b026fed26 )

**Comment by thutterer** [comment]
> Please merge only after #298 is merged and verified to pass the Maintenance

I just merged #298 into master, so you should be good to go to merge and package :ok_hand:

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/24658/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged aaec0169fb97e814f315ddb934e3895a37d4f8c6 into c5de78af9431bb24ffdc5d9c872f2fbb13a12aa7 )

**Comment by thutterer** [comment]
> you should be good to go to merge and package

Oh before you do, you probably should `git rebase master` on this branch. I think that gives a cleaner commit history than this :octocat: "Update branch" button above.

**Comment by digitaltom** [comment]
If you have time: I think it would be useful to add some output to the write_config method like: `Persisted options url, namespace to /etc/SUSEConnect`
What do you think?

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/25511/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged f2ea7afef19c3eefe52dec842efd137e3d08314c into c5de78af9431bb24ffdc5d9c872f2fbb13a12aa7 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/25514/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged a324f0b337fececf4e82fa9e17484df61618d8c1 into c5de78af9431bb24ffdc5d9c872f2fbb13a12aa7 )

**Pull request merged**